src/main/com/fulcrologic/fulcro/algorithms/normalize.cljc

```diff
@@ -60,16 +62,13 @@
                           (when-not (empty? is)
                             (swap! refs
                               (fn [refs]
-                                (reduce (fn [m [i x]]
-                                          (update-in m i merge x))
-                                  refs (zipmap is xs)))))
+                                (reduce (fn [m [i x]] (update-in m i merge x)) ; merge2data-entity
+                                  refs (map vector is xs)))))
```

Comment:
I don't want changes to this file. You can add a docstring if you want, and comments would be OK *if* they are clear to the reader....having zipmap change to map is unacceptable. I don't want any algorithmic changes at all.

I'd be ok with var renames, but not ones that alias outer things or collide with clojure.core names. So, for example, I'd be ok with changing `k` to `join-key` and `sel` to `subquery`.

Reply:
See my updated comment to the issue. I think that's actually the way to go (a function that calculates error after thrown)